mage_integrations/mage_integrations/destinations/salesforce/target_salesforce/session_credentials.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from dataclasses import dataclass
+import abc
+from typing import Union
+import requests
+from collections import namedtuple
+from simple_salesforce import SalesforceLogin
+
+
+LOGGER = logging.getLogger(__name__)
+
+OAuthCredentials = namedtuple(
+    "OAuthCredentials", ("client_id", "client_secret", "refresh_token")
+)
+
+PasswordCredentials = namedtuple(
+    "PasswordCredentials", ("username", "password", "security_token")
+)
+
+
+@dataclass
+class Session:
+    session_id: str
+    instance: str = None
+    instance_url: str = None
+
+
+def parse_credentials(config: dict) -> Union[OAuthCredentials, PasswordCredentials]:
+    for cls in reversed((OAuthCredentials, PasswordCredentials)):
+        creds = cls(*(config.get(key) for key in cls._fields))
+        if all(creds):
+            return creds
+
+    raise Exception(
+        "Cannot create credentials from config. Target supports OAuth and Password authentication."
+    )
+
+
+class SalesforceAuth(metaclass=abc.ABCMeta):
+    def __init__(self, credentials, domain):
```

Comment:
Can we pass the logger to this class as well?

Reply:
added on 940f3032c47cd0d717a20e0b1e94baf3bcd218a5